File: 345-reverse-vowels-of-a-string/reverse-vowels-of-a-string.cpp
class Solution {
public:
    string reverseVowels(string s) {
        string arr1 ={};
        int n = s.length();
        for(int i=0;i<n;i++){
            if (s[i] == 'a' || s[i] == 'A' || 
                s[i] == 'e' || s[i] == 'E' || 
                s[i] == 'i' || s[i] == 'I' || 
                s[i] == 'o' || s[i] == 'O' || 
                s[i] == 'u' || s[i] == 'U'){
                arr1.push_back(s[i]);
            }
        }
    int size = arr1.size();
    
    std::reverse(arr1.begin(), arr1.end());
    
            int j = 0;
            for(int i=0;i<s.length();i++){
            if (s[i] == 'a' || s[i] == 'A' || 
                s[i] == 'e' || s[i] == 'E' || 
                s[i] == 'i' || s[i] == 'I' || 
                s[i] == 'o' || s[i] == 'O' || 
                s[i] == 'u' || s[i] == 'U'){
                s[i]=arr1[j];
                j++;
            }
        }
    return s;
    }
    
};

File: 345-reverse-vowels-of-a-string/reverse-vowels-of-a-string.py
class Solution:
    def reverseVowels(self, s: str) -> str:

        vowels = "aeiouAEIOU"
        s = list(s)
        l = 0
        r = len(s)-1

        while l < r:
            if s[l] in vowels and s[r] in vowels:
                s[l],s[r] = s[r],s[l]
                l += 1
                r -= 1
            elif s[l] in vowels and s[r] not in vowels:
                r -= 1
            elif s[l] not in vowels and s[r] in vowels:
                l += 1
            else:
                l += 1
                r -= 1
        res = "".join(s)
        return res
        